File: integration-tests/hibernate-orm-graphql-panache/src/main/java/io/quarkus/it/hibertnate/orm/graphql/panache/GraphQLResource.java
package io.quarkus.it.hibertnate.orm.graphql.panache;

import java.util.List;

import javax.inject.Inject;

import org.eclipse.microprofile.graphql.Description;
import org.eclipse.microprofile.graphql.GraphQLApi;
import org.eclipse.microprofile.graphql.Query;

import io.quarkus.logging.Log;

@GraphQLApi
public class GraphQLResource {

    @Inject
    BookRepository bookRepository;

    @Query("authors")
    @Description("Retrieve the stored authors")
    public List<Author> getAuthors() {
        Log.info("Getting all authors");
        return Author.listAll();
    }

    @Query("books")
    @Description("Retrieve the stored books")
    public List<Book> getBooks() {
        Log.info("Getting all books");
        return bookRepository.listAll();
    }
}

File: integration-tests/kotlin-maven-invoker/src/test/resources/projects/kotlin-compiler-args/src/main/kotlin/org/acme/GraphQLResource.kt
package org.acme

import org.eclipse.microprofile.graphql.GraphQLApi
import org.eclipse.microprofile.graphql.NonNull
import org.eclipse.microprofile.graphql.Query

@GraphQLApi
class GraphQLResource {

    @Query
    @NonNull
    fun getBananas(): List<@NonNull Banana> {
        return listOf(Banana("yellow", 5), Banana("green", 3))
    }
}

class Banana(
    val color: String,
    val size: Int,
)
